src/h5web/visualizations/containers/NxSpectrumContainer.tsx

```diff
@@ -30,16 +19,13 @@ function NxSpectrumContainer(props: VisContainerProps): ReactElement {
     throw new Error('NXdata group not found');
   }
 
-  const signalName = getAttributeValue(nxDataGroup, 'signal');
-  assertStr(signalName);
+  const nxData = useParsedNxData(nxDataGroup, metadata);
 
-  const signalDataset = getLinkedEntity(signalName, nxDataGroup, metadata);
-
-  if (!signalDataset || !isDataset(signalDataset)) {
+  if (!nxData) {
     throw new Error('Signal dataset not found');
```

Comment:
Hmm if `nxData` is `undefined`, it means the values are being fetched, doesn't it? So you should return `<></>` rather than trow, maybe?

Reply:
Good point but apparently, the rendering waits for the fetching ? :shrug: 

Also, I cannot return here as there is the `useState` hook below whose initial value depends on `nxData.signal.dims`...

Reply:
I think we don't see any error because `Suspense` kind of hides it. The function finishes executing, so throwing does not seem like a good idea. Hmm this is quite a problem, in my opinion.

Reply:
Perhaps `useNxData` should return what it can the first time around just so the `useState` can be properly initialized (and the `return` moved below the `useState`.